### PR TITLE
Renamed ProgramInfo to ExecutionDetails 

### LINKED ISSUE #743: Duplicated `ProgramInfo`
We have a duplicated struct `ProgramInfo`. This is undesirable as it might create confusion regarding its definition and references.

Core:

https://github.com/0xPolygonMiden/miden-vm/blob/159a04f3d4648f77d664642409e521438378bd6c/core/src/program/info.rs#L14-L18

Miden:

https://github.com/0xPolygonMiden/miden-vm/blob/159a04f3d4648f77d664642409e521438378bd6c/miden/src/tools/mod.rs#L55-L60

Ideally, the onde defined in `miden` should be renamed, as it will cause smaller impact than the one defined in `core`. The one in `core` is used to verify stark proofs, while the one in `miden` is used solely to provide user output.

### PR DESCRIPTION
**Pull request by hackaugusto**
## Describe your changes

fixes #743

## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: miden/src/tools/mod.rs [hunk 1]
```diff
@@ -34,32 +34,31 @@ impl Analyze {
         let stack_inputs = input_data.parse_stack_inputs()?;
         let advice_provider = input_data.parse_advice_provider()?;
 
-        let program_info: ProgramInfo = analyze(program.as_str(), stack_inputs, advice_provider)
-            .expect("Could not retrieve program info");
+        let program_info: ExecutionDetails =
```
Comment (vlopes11) [inline miden/src/tools/mod.rs:37]: optional: maybe use `execution_details` as variable name to keep it consistent with the type?

**Review by vlopes11 — APPROVED** [review]
Looks good! Small optional nit

File: miden/src/tools/mod.rs (hunk 1 above)
Comment (hackaugusto) [reply miden/src/tools/mod.rs]: ofcourse! done :)

File: miden/src/tools/mod.rs [hunk 2]
```diff
@@ -34,32 +34,31 @@ impl Analyze {
...
+// EXECUTION DETAILS
 // ================================================================================================
 
-/// Contains info of a program. Used for program analysis. Contains the following fields:
-/// - total_vm_cycles: vm cycles it takes to execute the entire program
-/// - total_noops: total noops executed as part of a program
-/// - asm_op_stats: vector of [AsmOpStats] that contains assembly instructions and
-///   the number of vm cycles it takes to execute the instruction and the number of times the
-///   instruction is run as part of the given program.
+/// Contains details of the exeuction of a program, used for program analysis
```
Comment (bobbinth) [inline miden/src/tools/mod.rs:50]: minor nit: I'd change this to "Contains details of executing a program, used for program analysis."

File: miden/src/tools/mod.rs [hunk 3]
```diff
@@ -242,9 +241,9 @@ mod tests {
             "proc.foo.1 loc_store.0 end begin mem_storew.1 dropw push.17 push.1 movdn.2 exec.foo end";
         let stack_inputs = StackInputs::default();
         let advice_provider = MemAdviceProvider::default();
-        let program_info = super::analyze(source, stack_inputs, advice_provider)
+        let execution_details = super::analyze(source, stack_inputs, advice_provider)
             .expect("analyze_test: Unexpected Error");
-        let expected_program_info = ProgramInfo {
+        let expected_program_info = ExecutionDetails {
```
Comment (bobbinth) [inline miden/src/tools/mod.rs:246]: nit: I'd probably rename the variable to `expected_details` or something like that.

File: miden/src/tools/mod.rs [hunk 4]
```diff
@@ -34,32 +34,31 @@ impl Analyze {
         let stack_inputs = input_data.parse_stack_inputs()?;
         let advice_provider = input_data.parse_advice_provider()?;
 
-        let program_info: ProgramInfo = analyze(program.as_str(), stack_inputs, advice_provider)
-            .expect("Could not retrieve program info");
+        let execution_details: ExecutionDetails =
+            analyze(program.as_str(), stack_inputs, advice_provider)
+                .expect("Could not retrieve program info");
```
Comment (bobbinth) [inline miden/src/tools/mod.rs:39]: nit: I'd probably update the `expect` message.

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left a couple of non-blocking comments inline.

Also, there is a reference to program info on line 91 which should probably be updated.

**Pull request merged**